### PR TITLE
mnist examples doc

### PR DESCRIPTION
**Pull request by QuanluZhang**

### CODE REVIEW

**Comment by QuanluZhang** [comment]
@PurityFan please help review

File: docs/mnist_examples.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+`code directory: nni/examples/trials/mnist-hyperband/`
+
+<a name="mnist-nested"></a>
+**mnist -- tuning within a nested search space**
+
+This example is to show that NNI also support nested search space. The search space file is an example of how to define nested search space.
+
+`code directory: nni/examples/trials/mnist-cascading-search-space/`
+
+<a name="mnist-kuberflow-tf"></a>
```
Comment (SparkSnail) [inline docs/mnist_examples.md]: mnist-kubeflow-tf

File: docs/mnist_examples.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+**distributed mnist (tensorflow) using kubeflow**
+
+This example is to show how to run distributed training on kubeflow through NNI. Users can simply provide distributed training code and a configure file which specifies the kubeflow mode. For example, what is the command to run ps and what is the command to run worker, and how many resources they consume. This example is implemented in tensorflow, thus, uses kubeflow tensorflow operator.
+
+`code directory: nni/examples/trials/mnist-distributed/`
+
+<a name="mnist-kuberflow-pytorch"></a>
```
Comment (SparkSnail) [inline docs/mnist_examples.md]: mnist-kubeflow-pytorch

File: docs/mnist_examples.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
+# MNIST examples
+
+CNN MNIST classifier for deep learning is similar to `hello world` for programming languages. Thus, we use MNIST as example to introduce different features of NNI. The examples are listed below:
```
[the remark this answers is not in the document]
Comment (QuanluZhang) [reply docs/mnist_examples.md]: MNIST is only a dataset, the name is in capital.

[the remark this answers is not in the document]
Comment (QuanluZhang) [reply docs/mnist_examples.md]: yes, but mnist is more well known, though it is only a dataset.

File: docs/mnist_examples.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+ - [mnist with NNI annotation](#mnist-annotation)
+ - [mnist in keras](#mnist-keras)
+ - [mnist -- tuning with batch tuner](#mnist-batch)
+ - [mnist -- tuning with hyperband](#mnist-hyperband)
+ - [mnist -- tuning within a nested search space](#mnist-nested)
+ - [distributed mnist (tensorflow) using kubeflow](#mnist-kubeflow-tf)
+ - [distributed mnist (pytorch) using kubeflow](#mnist-kubeflow-pytorch)
+
+<a name="mnist"></a>
+**mnist**
```
[the remark this answers is not in the document]
Comment (QuanluZhang) [reply docs/mnist_examples.md]: good suggestion. will fixed

File: docs/mnist_examples.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+This example is implemented in keras. It is also a network for MNIST dataset, with two convolution layers, one pooling layer, and two fully connected layers.
+
+`code directory: nni/examples/trials/mnist-keras/`
+
+<a name="mnist-batch"></a>
+**mnist -- tuning with batch tuner**
+
+This example is to show how to use batch tuner. Users simply list all the configurations they want to try in the search space file. NNI will try all of them.
+
+`code directory: nni/examples/trials/mnist-batch-tune-keras/`
```
[the remark this answers is not in the document]
Comment (QuanluZhang) [reply docs/mnist_examples.md]: yes, will fix

File: docs/mnist_examples.md (hunk 1 above)
Comment (QuanluZhang) [reply docs/mnist_examples.md]: yes, thanks.

File: docs/mnist_examples.md (hunk 2 above)
Comment (QuanluZhang) [reply docs/mnist_examples.md]: yes, thanks

File: docs/mnist_examples.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (QuanluZhang) [reply docs/mnist_examples.md]: done, thanks!

**Pull request merged**